src/store/getters.ts

```diff
@@ -0,0 +1,10 @@
+import type { OverallState } from './types'
+
+export const usersById = (state: OverallState) => state.users
+
+export const allUsers = (state: OverallState) => Object.values(state.users)
+
+export const allPosts = (state: OverallState) =>
+  state.posts.sort((a, b) => new Date(b.createDate).getTime() - new Date(a.createDate).getTime())
```

Comment:
What kind of logic should be appleid to getters?
When should logic be applied to a getter and when should it be calculated in a `setup`?

For example, would it be advisable / possible to have a getter that accepts a `userId` argument, so that it can return `state.users[userId]`

Reply:
Can be complicated but no mutations to state
If it's too complicated, might rethink what has been received from backend

> For example, would it be advisable / possible to have a getter that accepts a userId argument, so that it can return state.users[userId]
just calculate in `computed`, as to preserve caching func of getters